pkg/network/server.go

```diff
@@ -125,11 +125,14 @@ type (
 		// lastRequestedHeader contains a height of the last requested header.
 		lastRequestedHeader atomic.Uint32
 
-		register   chan Peer
-		unregister chan peerDrop
-		handshake  chan Peer
-		quit       chan struct{}
-		relayFin   chan struct{}
+		register       chan Peer
+		unregister     chan peerDrop
+		handshake      chan Peer
+		quit           chan struct{}
+		relayFin       chan struct{}
+		runFin         chan struct{}
+		broadcastTxFin chan struct{}
+		runProtoFin    chan struct{}
```

Comment:
Yep, let's also wait for all `txHandlerLoop` routines with a waitgroup since all of them has draining loops which must be awaited on server exit.